Analysis/SkimEvents.py

```diff
@@ -0,0 +1,102 @@
+import ROOT
+import os
+import shutil
+ROOT.gROOT.SetBatch(True)
+ROOT.EnableThreadSafety()
+
+
+def ListToVector(list, type="string"):
+	vec = ROOT.std.vector(type)()
+	for item in list:
+		vec.push_back(item)
+	return vec
+
+col_type_dict = {
+  'Float_t':'float',
+  'Bool_t':'bool',
+  'Int_t' :'int',
+  'ULong64_t' :'unsigned long long',
+  'Long_t' :'long',
+  'UInt_t' :'unsigned int',
+  'ROOT::VecOps::RVec<float>':'ROOT::VecOps::RVec<float>',
+  'ROOT::VecOps::RVec<int>':'ROOT::VecOps::RVec<int>'
+  }
+
+def make_df(inputFileCentral,inputFileShifted,outFile,treeName):
+  df_out = ROOT.RDataFrame('Events', inputFileShifted)
+  colNames = [str(c) for c in df_out.GetColumnNames()]
+  entryIndexIdx = colNames.index("entryIndex")
+  colNames[entryIndexIdx], colNames[0] = colNames[0], colNames[entryIndexIdx]
+  col_types = [str(df_out.GetColumnType(c)) for c in colNames]
+  tuple_maker = ROOT.analysis.TupleMaker(*col_types)('Events', inputFileCentral, 4)
+  print("tuplemaker created")
+  df_out = tuple_maker.processOut(ROOT.RDF.AsRNode(df_out))
+  print("tuplemaker processed")
+  df_out = df_out.Define("isValid", "_entryCentral.use_count() > 0")
+  df_unique = df_out.Filter("!isValid")
+  df_out_valid = df_out.Filter('isValid')
+  colToSave_diff= []
+  colToNotToMakeDiff=  ["period","run", "sample_name", "sample_type", "channelId", "entryIndex", "event", "isData", "luminosityBlock"]
+  colToSave_noDiff= [ "entryIndex"]
+
+  condition_noDiff_list = []
+  condition_Valid_list = []
+
+  for var_idx,var_name in enumerate(colNames):
+    if var_name in colToNotToMakeDiff: continue
+    condition_noDiff_list.append(f"analysis::IsSame(_entryCentral->GetValue<{col_type_dict[col_types[var_idx]]}>({var_idx}),{var_name})")
+    #condition_Valid_list.append(f"! analysis::IsSame(_entryCentral->GetValue<{col_type_dict[col_types[var_idx]]}>({var_idx}),{var_name})")
+
+  condition_noDiff = ' && '.join(condition_noDiff_list)
+  df_out_valid = df_out_valid.Define("isSame", condition_noDiff)
+  df_out_valid_noDiff = df_out_valid.Filter("isSame")
+  df_out_valid_diff = df_out_valid.Filter("!isSame")
+  for var_idx,var_name in enumerate(colNames):
+    if var_name in colToNotToMakeDiff: continue
+    df_out_valid_diff=df_out_valid_diff.Define(f"diff_{var_name}", f"""analysis::Delta(_entryCentral->GetValue<{col_type_dict[col_types[var_idx]]}>({var_idx}),{var_name})""")
```

Comment:
`{var_name}Diff` to make the naming convention "nano-friendly"

Reply:
ok